lib/rabblebot/plugins/weather.rb

```diff
@@ -0,0 +1,86 @@
+require 'open-uri'
+require 'json'
+
+class RabbleBot
+  attr_reader :google_api, :forecastio_api
+  
+  module RabbleBotPlugin
+    # a weather plugin
+    class Weather < BasicPlugin
+      def initialize(bot)
+        super(bot)
+        @bot.info 'Loading Weather Plugin'
+        @google_api = 'YOUR_GOOGLEAPI_KEY'
+        @forecastio_api = 'YOUR_FORECAST_IO_API_KEY'
+        add_weather_handler
+        @bot.info 'Weather Plugin loaded!'
+      end
+      
+      def weather_help(e)
+        e.respond '/weather ZIP - tells you what the weather of ZIP is.'
+      end
+
+      def weather_query(e, message)
+        gmaps_json_response = open("https://maps.googleapis.com/maps/api/geocode/json?address=#{message}&key=#{@google_api}").read
+        gmaps_json = JSON.parse(gmaps_json_response)
```

Comment:
You might want to catch `JSON::ParserError` or `JSON::JSONError` and return an error message in case SomethingGoesWrong™